### PR TITLE
Document how to resolve `cquery --output=files` paths

### LINKED ISSUE #17658: cquery --output=files is not useful for external dependencies
### Description of the feature request:

With https://github.com/bazelbuild/bazel/pull/15979 `cquery --output=files` provides a relative path to the output or source file.  The feature request is to make the output as readily usable when the target is from an external dependency.

Bazel 6.0.0

```
% cat WORKSPACE
workspace(name = "com_example")
load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
% touch BUILD
% bazel cquery --output=files @bazel_tools//tools/build_defs/repo:git.bzl
Starting local Bazel server and connecting to it...
INFO: Analyzed target @bazel_tools//tools/build_defs/repo:git.bzl (1 packages loaded, 1 target configured).
INFO: Found 1 target...
external/bazel_tools/tools/build_defs/repo/git.bzl
INFO: Elapsed time: 3.607s, Critical Path: 0.00s
INFO: 0 processes.
INFO: Build completed successfully, 0 total actions
```

There is no WORKSPACE-relative "external" directory.

I happen to know that I can currently use `$(bazel info output_path)/..` to get the parent directory to "external", but that's arcane knowledge and isn't directly provided by the cquery feature.

### What underlying problem are you trying to solve with this feature?

Have `cquery --output=files` behave consistently for all files known to bazel.

### Which operating system are you running Bazel on?

Linux

### What is the output of `bazel info release`?

release 6.0.0

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.

_No response_

### What's the output of `git remote get-url origin; git rev-parse master; git rev-parse HEAD` ?

_No response_

### Have you found anything relevant by searching the web?

Linked issues and PRs associated with https://github.com/bazelbuild/bazel/pull/15979

### Any other information, logs, or outputs that you want to share?

_No response_

### PR DESCRIPTION
**Pull request by fmeum**
Fixes #17658

### CODE REVIEW

**Comment by fmeum** [comment]
@jscheid-ventana Could you give this a read and say whether you find it understandable?

File: site/en/query/cquery.md [hunk 1]
```diff
@@ -368,6 +368,12 @@ output groups as determined by the
 [`--output_groups`](/reference/command-line-reference#flag--output_groups) flag.
 It does include source files.
 
+All paths emitted by this output format are relative to the
+[execroot](https://bazel.build/remote/output-directories), which can be obtained
+via `bazel info execution_root`. Paths to files in the main repository also
+resolve relative to the project directory (assuming the `bazel-out` convenience
```
Comment (jscheid-ventana) [inline site/en/query/cquery.md:374]: Move parenthetical to prefix "if" clause?

Is "project directory" the right technical term, vs "workspace directory"?

Comment (jscheid-ventana) [reply site/en/query/cquery.md:374]: Otherwise, looks good.  Thanks!

**Review by gregestren — COMMENTED** [review]
Happy to approve after @jscheid-ventana 's comments resolved.

**Review by gregestren — APPROVED** [review]

**Pull request closed without merging**